### ISSUE TITLE
Plans for ragged tensor support

### ISSUE THREAD

**Issue by foxik** [keras-team/keras-core#456]
Dear developers,

I very much appreciate the effort of supporting multiple backends, including cross-backend usage of models, and the possibility of using `tf.data` in PyTorch&JAX.

I am also a frequent user of ragged tensors, which I find extremely useful (I am an NLP). Do you think they will be included in the Keras 3.0 release? In an ideal world they would be supported in all backends (by using masked tensors in PyTorch and JAX, considering that they do not have the "real" ragged implementation), but my realistic guess is that ragged tensors will be TF-specific? Are there plans to support them in losses, metrics, Dense and RNN layers, Ops, ...?

Thanks!

**Comment by foxik** [comment]
Thanks for your fast and clear response!

If `tf.keras` eventually is Keras Core (as mentioned on https://keras.io/keras_core/announcement/), then not supporting ragged tensors would be a considerable drop of functionality (and it would not be easy to use ragged tensors in high-level TF code afterwards); on the other hand I understand that the effort to support ragged tensors is substantial -- it needs to be (and currently is) supported in losses, metrics, graph inputs and outputs (that means that even `model.predict` might sometimes return a `tf.RaggedTensor` instead of an `np.array`), layers, ...
